File: client/src/main/java/org/ntcrckr/ppidpo/client/service/ClientConfig.java
package org.ntcrckr.ppidpo.client.service;

import org.ntcrckr.ppidpo.client.service.impl.ClientImpl;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ClientConfig {
    @Bean
    public CommandLineRunner run(Client client) {
        return args -> client.run();
    }

    @Bean
    public Client client() {
        return new ClientImpl("../server_input_folder");
    }
}

File: client/src/main/kotlin/org/ntcrckr/ppidpo2/rest/ClientConfig.kt
package org.ntcrckr.ppidpo2.rest

import org.springframework.boot.web.client.RestTemplateBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestTemplate

@Configuration
class ClientConfig {
    @Bean
    fun restTemplate(builder: RestTemplateBuilder): RestTemplate = builder.build()

    @Bean
    fun infinitePoller(restTemplate: RestTemplate): InfinitePoller = InfinitePoller(
        restTemplate,
        "http://localhost:8080/currencyRate",
    )
}
